### PR TITLE
PHP Bookworm Image Fixes

### PR DESCRIPTION
**Pull request by prathambande**
The purpose of this PR is to fix the existing issues which existed in the PHP Bookworm images.

- Changed bookworm docker file to use Debian/12 instead of Debian/11
- Pinned the versions of the package which is found in bookworm images

Link for the PHP bookworm artifacts generated: [Pipeline](https://dev.azure.com/msazure/Antares/_build/results?buildId=97860221&view=results)

### CODE REVIEW

**Review by SarathBussa — DISMISSED** [review]

File: images/runtime/php-fpm/8.3/bookworm.Dockerfile [hunk 1]
```diff
@@ -26,9 +26,11 @@ RUN set -eux \
 		gnupg2 \
 		apt-transport-https \
 	&& curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
-	&& curl https://packages.microsoft.com/config/debian/11/prod.list > /etc/apt/sources.list.d/mssql-release.list \
+	&& curl https://packages.microsoft.com/config/debian/12/prod.list > /etc/apt/sources.list.d/mssql-release.list \
+	&& curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor -o /usr/share/keyrings/microsoft-prod.gpg \
 	&& apt-get update \
-	&& ACCEPT_EULA=Y apt-get install -y msodbcsql17 msodbcsql18=18.1.2.1-1 odbcinst1debian2=2.3.7 odbcinst=2.3.7 unixodbc=2.3.7 unixodbc-dev=2.3.7
+	&& ACCEPT_EULA=Y apt-get install -y msodbcsql17=17.10.6.1-1 msodbcsql18=18.3.3.1-1 odbcinst1debian2=2.3.11-2+deb12u1 odbcinst=2.3.11-2+deb12u1 unixodbc=2.3.11-2+deb12u1 unixodbc-dev=2.3.11-2+deb12u1
```
Comment (sarsharma) [inline images/runtime/php-fpm/8.3/bookworm.Dockerfile:32]: unixodbc-dev=2.3.11-2+deb12u1
odbcinst1debian2=2.3.11-2+deb12u1 

are these stable packages, what does `+deb12u1` in their version indicate?

File: images/runtime/php/8.3/bookworm.Dockerfile [hunk 2]
```diff
@@ -27,9 +27,11 @@ RUN set -eux \
 		gnupg2 \
 		apt-transport-https \
 	&& curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
-    && curl https://packages.microsoft.com/config/debian/11/prod.list > /etc/apt/sources.list.d/mssql-release.list \
+    && curl https://packages.microsoft.com/config/debian/12/prod.list > /etc/apt/sources.list.d/mssql-release.list \
+	&& curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor -o /usr/share/keyrings/microsoft-prod.gpg \
 	&& apt-get update \
-	&& ACCEPT_EULA=Y apt-get install -y msodbcsql17 msodbcsql18=18.1.2.1-1 odbcinst1debian2=2.3.7 odbcinst=2.3.7 unixodbc=2.3.7 unixodbc-dev=2.3.7
+	&& ACCEPT_EULA=Y apt-get install -y msodbcsql17=17.10.6.1-1 msodbcsql18=18.3.3.1-1 odbcinst1debian2=2.3.11-2+deb12u1 odbcinst=2.3.11-2+deb12u1 unixodbc=2.3.11-2+deb12u1 unixodbc-dev=2.3.11-2+deb12u1
+	# && ACCEPT_EULA=Y apt-get install -y msodbcsql17 msodbcsql18=18.1.2.1-1 odbcinst1debian2=2.3.7 odbcinst=2.3.7 unixodbc=2.3.7 unixodbc-dev=2.3.7
```
Comment (sarsharma) [inline images/runtime/php/8.3/bookworm.Dockerfile:34]: nit: remove?

File: images/runtime/php-fpm/8.3/bookworm.Dockerfile [hunk 3]
```diff
@@ -26,9 +26,11 @@ RUN set -eux \
...
+	# && ACCEPT_EULA=Y apt-get install -y msodbcsql17 msodbcsql18=18.1.2.1-1 odbcinst1debian2=2.3.7 odbcinst=2.3.7 unixodbc=2.3.7 unixodbc-dev=2.3.7
```
Comment (sarsharma) [inline images/runtime/php-fpm/8.3/bookworm.Dockerfile:33]: nit: remove commented line

File: images/runtime/php-fpm/8.3/bookworm.Dockerfile (hunk 1 above)
Comment (prathambande) [reply images/runtime/php-fpm/8.3/bookworm.Dockerfile:32]: Yes, these packages seem to be stable (the change log says so, [odbcinst](https://debian.pkgs.org/12/debian-main-amd64/odbcinst_2.3.11-2+deb12u1_amd64.deb.html) , [odbcinst1](https://debian.pkgs.org/12/debian-main-amd64/odbcinst1debian2_2.3.11-2+deb12u1_amd64.deb.html) , [unixodbc](https://debian.pkgs.org/12/debian-main-amd64/unixodbc_2.3.11-2+deb12u1_amd64.deb.html)

**+deb12u1**  indicates first update to the package in Debian 12

**Review by SarathBussa — APPROVED** [review]

**Review by sarsharma — APPROVED** [review]

**Pull request merged**